geode-core/src/test/java/com/gemstone/gemfire/security/IntegratedSecurityCacheLifecycleDistributedTest.java

```diff
@@ -0,0 +1,151 @@
+import com.gemstone.gemfire.cache.client.ClientCache;
+import com.gemstone.gemfire.cache.client.ClientCacheFactory;
+import com.gemstone.gemfire.cache.client.ClientRegionShortcut;
+import com.gemstone.gemfire.cache.server.CacheServer;
+import com.gemstone.gemfire.distributed.internal.InternalLocator;
+import com.gemstone.gemfire.internal.AvailablePort;
+import com.gemstone.gemfire.management.internal.security.JSONAuthorization;
+import com.gemstone.gemfire.security.templates.UserPasswordAuthInit;
+import com.gemstone.gemfire.test.dunit.DistributedTestUtils;
+import com.gemstone.gemfire.test.dunit.Host;
+import com.gemstone.gemfire.test.dunit.NetworkUtils;
+import com.gemstone.gemfire.test.dunit.VM;
+import com.gemstone.gemfire.test.dunit.cache.internal.JUnit4CacheTestCase;
+import com.gemstone.gemfire.test.junit.categories.DistributedTest;
+import com.gemstone.gemfire.test.junit.categories.SecurityTest;
+
+import org.assertj.core.api.ThrowableAssert.ThrowingCallable;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.mockito.Spy;
+
+@Category({DistributedTest.class, SecurityTest.class})
+public class IntegratedSecurityCacheLifecycleDistributedTest extends JUnit4CacheTestCase {
+
+  private static SpySecurityManager spySecurityManager;
+
+  private VM locator;
+
+  @Override
+  public final void postSetUp() throws Exception {
+    Host host = Host.getHost(0);
+    locator = host.getVM(0);
+    JSONAuthorization.setUpWithJsonFile("clientServer.json");
+    int locatorPort = AvailablePort.getRandomAvailablePort(AvailablePort.SOCKET);
+    String locators =  NetworkUtils.getServerHostName(host) + "[" + locatorPort + "]";
+
+    spySecurityManager = new SpySecurityManager();
+
+    locator.invoke(() -> {
+      spySecurityManager = new SpySecurityManager();
+      DistributedTestUtils.deleteLocatorStateFile(locatorPort);
+
+      final Properties properties = new Properties();
+      properties.setProperty(MCAST_PORT, "0");
+      properties.setProperty(START_LOCATOR, locators);
+      properties.setProperty(SECURITY_MANAGER, SpySecurityManager.class.getName()+".create");
+      properties.setProperty(USE_CLUSTER_CONFIGURATION, "false");
+      getSystem(properties);
+      getCache();
+    });
+
+    final Properties properties = new Properties();
+    properties.setProperty(MCAST_PORT, "0");
+    properties.setProperty(SECURITY_MANAGER, SpySecurityManager.class.getName()+".create");
+    properties.setProperty(LOCATORS, locators);
+    properties.setProperty(USE_CLUSTER_CONFIGURATION, "false");
+    getSystem(properties);
+
+    CacheServer server1 = getCache().addCacheServer();
+    server1.setPort(0);
+    server1.start();
+
+    getCache();
+  }
+
+  @Test
+  public void initAndCloseTest () {
+    locator.invoke(() -> {
+      verifyInitInvoked();
+    });
+    verifyInitInvoked();
+    getCache().close();
+    verifyCloseInvoked();
+    locator.invoke(() -> {
+      getCache().close();
```

Comment:
Can we put all the locator.invoke in one block? like this:
locator.invoke(()->{
   verifyInitInvoked();
   getCache().close();
   verifyCloseInvoked();
});